presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsMetadata.java

```diff
@@ -254,8 +257,15 @@ public static Type getPrestoType(ColumnType tpcdsType)
                 return DateType.DATE;
             case DECIMAL:
                 return createDecimalType(tpcdsType.getPrecision().get(), tpcdsType.getScale().get());
-            case CHAR:
-                return createCharType(tpcdsType.getPrecision().get());
+            case CHAR: {
+                if (isNativeExecution) {
+                    // Presto native does not support CHAR type yet.
```

Comment:
This is not correct. We cannot hide incompatibilities in this way. Let's use a cast explicitly.

Reply:
Let's also document in the RFC that the native tpcds connector differs from the Java implementation for this type.